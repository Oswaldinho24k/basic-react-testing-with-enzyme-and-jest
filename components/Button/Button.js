import PropTypes from 'prop-types';

/**
 * Component for showing details of the user.
 * 
 * @component
 * @example
 * 
 * const handleClick = () => console.log('clicked')
 * return (
 *   <Button disabled={false} onClick={handleClick}> The Button </Button>
 * )
 */

const Button = ({children, disabled, onClick}) => {
  return(
    <button disabled={disabled} onClick={onClick}>
      {children}
    </button>
  )
}

Button.defaultProps = {
  disabled: false
}

Button.propTypes = {
  /**
   * Buttons's disabled status
   */
  disabled: PropTypes.bool,
  /**
   * Buttons's onClick function
   */
  onClick: PropTypes.func,
  /**
   * Buttons's children text
   */
  children: PropTypes.any.isRequired
}

export default Button
