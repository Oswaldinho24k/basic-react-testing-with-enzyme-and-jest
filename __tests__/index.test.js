import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import Index from "../pages/index.js";
import Button from "../components/Button/Button.js";

describe("With Enzyme", () => {
  test('Component shows Kiubo', () => {
    const component = shallow(<Index />);
    expect(component.find("h2").text()).toEqual("Kiubole");
  });
});

describe("With Snapshot Testing", () => {
  test('App shows "Kiubo"', () => {
    const component = renderer.create(<Index />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("show text after click", () => {
  test('Show quedate en tu casa', () => {
    const component = shallow(<Index />);
    component.find(Button).at(1).simulate('click');
    console.log(component.find("p").text())
    expect(component.find("p").text()).toEqual("Qu√©date en tu casa!!!");
  });
});