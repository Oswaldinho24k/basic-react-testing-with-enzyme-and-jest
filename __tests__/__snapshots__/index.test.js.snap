// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Kiubo" 1`] = `
<div>
  <h2>
    Kiubole
  </h2>
  <p>
    
  </p>
  <section>
    <button
      disabled={true}
      onClick={[Function]}
    >
       Disabled
    </button>
    <button
      disabled={false}
      onClick={[Function]}
    >
       PÃ­cale
    </button>
  </section>
</div>
`;
